presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestJoinQueries.java

```diff
@@ -2392,6 +2392,12 @@ private void testJoinsWithNullsInternal(Session session)
                 "SELECT * FROM (VALUES 2, 3, null) a(x) " +
                         "FULL OUTER JOIN (VALUES 3, 4, null) b(x) ON a.x = b.x WHERE a.x IS NULL",
                 "SELECT * FROM VALUES (NULL, 4), (NULL, NULL), (NULL, NULL)");
+
+        assertQuerySucceeds(
+                session,
+                "SELECT * FROM customer C INNER JOIN " +
+                        "(SELECT SPLIT(comment, '.') AS k FROM part) P " +
+                        "ON C.n = P.k[2] WHERE CARDINALITY(P.k) >= 2");
```

Comment:
Is this a repro case? Also, cast is another 

CAST(x as int) = y
WHERE x <> ''

Reply:
@kaikalur Yes, it's a repro case. I can add CAST as second test as well.